feat: improve error handling

NguyenHoangSon96 opened the pull request:
Closes #

## Proposed Changes

_Briefly describe your proposed changes:_

## Checklist

<!-- Checkboxes below this note can be erased if not applicable to your Pull Request. -->

- [ ] CHANGELOG.md updated
- [ ] Rebased/mergeable
- [ ] A test has been added if appropriate
- [ ] Tests pass
- [ ] Commit messages are [conventional](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Sign [CLA](https://www.influxdata.com/legal/cla/) (if not already signed)

karel-rehor reviewed (commented):
Looking at #134 I think the new classes `InfluxdDBClientError` and `InfluxdbClientQueryError` are a good start, but oclyke is looking for a taxonomy of more specific error definitions.  Also class names could include the version 3 value to make sure they are distinct from other influxdb libraries.  

Compare this to the Arrow Flight taxonomy of error classes.  (https://arrow.apache.org/docs/python/generated/pyarrow.flight.FlightError.html).

Firstly from oclyke's comment I think the user does not want to see these Arrow Flight errors or to deal with them.  

Secondly also from oclyke's comment I think the user wants to see an error class that is self documenting or self explanatory so that it is readily understood.

Thirdly we could implement a taxonomy similar to that of Arrow, which would wrap specific Arrow exceptions and their contents.  

e.g. 
```python 
class InfluxDB3ClientQueryServerError(InfluxDB3ClientError):
...
class InfluxDB3ClientQueryServerUnkownDBError(InfluxDB3ClientQueryServerError):
...
class InfluxDB3ClientQueryUnauthenticatedError(InfluxDB3ClientError):
...
class InfluxDB3ClientQueryUnavailableError(InfluxDB3ClientError):
...
etc.
```
We could then maybe have an `ArrowErrorHandler` class with a static method to map caught Arrow errors to classes in our error taxonomy

karel-rehor reviewed (changes requested):
1. An item for a more finely defined error taxonomy across all client libraries has been opened in the backlog.  So this PR represents only a partial solution to #140
2. 1 older unit test was failing locally, so I've updated it.  You may want to review or revert these changes.  
3. Some refactoring comments
    1. Since there already exists an [InfluxDBError](https://github.com/InfluxCommunity/influxdb3-python/blob/main/influxdb_client_3/write_client/client/exceptions.py) class in the `write_client` package, I think this needs to be taken into account.
    2. Since  `InfluxdbClientQueryError` only deals with _query_ errors, it could be declared in the `query` package
    3. another alternative might be to declare all exceptions/errors in a new `exceptions` package.

NguyenHoangSon96 commented:
> 1. An item for a more finely defined error taxonomy across all client libraries has been opened in the backlog.  So this PR represents only a partial solution to [feat: improve error handling #140](https://github.com/InfluxCommunity/influxdb3-python/pull/140)
> 2. 1 older unit test was failing locally, so I've updated it.  You may want to review or revert these changes.
> 3. Some refactoring comments
>    
>    1. Since there already exists an [InfluxDBError](https://github.com/InfluxCommunity/influxdb3-python/blob/main/influxdb_client_3/write_client/client/exceptions.py) class in the `write_client` package, I think this needs to be taken into account.
>    2. Since  `InfluxdbClientQueryError` only deals with _query_ errors, it could be declared in the `query` package
>    3. another alternative might be to declare all exceptions/errors in a new `exceptions` package.
> 
> BTW. When running unit tests locally, even though all tests now pass, once they are finished, I keep running across a threading issue associated with `urllib3` after the tests complete. I don't recall seeing this in the past. Needs further investigation.
> 
> ```
> ...
> RuntimeError: can't create new thread at interpreter shutdown
> The retriable error occurred during request. Reason: '<urllib3.connection.HTTPSConnection object at 0x71b678a54b90>: Failed to resolve 'none' ([Errno -3] Temporary failure in name resolution)'.
> The retriable error occurred during request. Reason: '<urllib3.connection.HTTPSConnection object at 0x71b678a57e00>: Failed to resolve 'none' ([Errno -3] Temporary failure in name resolution)'.
> ...
> ```

Hi @karel-rehor 
I have moved all exceptions to the exception package
Regarding the InfluxDBError class I don't want to touch it because of backward compatibility
And I changed all exception names to include the number 3
